### ISSUE TITLE
Proxy authentication needs custom headers and JWT

### ISSUE THREAD

**Issue by zkhezane** [Azure/azure-sdk-for-java#37054]
**Query/Question**

Hello,
Currently, we are calling our Azure Data Lake through the Rest API, but we want to change this and leverage Azure Java SDK to call our Azure Data Lake.

We are doing a POC to check if this is possible since we're using a Proxy to reach our Datalake, but we're blocked 
In the past we used to pass the Proxy information throught the headers oh the call as follows 

`headers = {
        'Proxy-Authorization': 'Bearer ' + jwt_conn, // this is the proxy token `
`
        "proxy_custom_parameter": "proxy_custom_parameter_value"
`
`
        "Authorization": f"Bearer {storage_token}" // Azure token
    }`
When we try to do that in Java code, it looks like follows :
```Java
   ProxyOptions proxyOptions = new ProxyOptions(ProxyOptions.Type.HTTP, new InetSocketAddress(pro, 20003));
        HttpClient httpClient = new NettyAsyncHttpClientBuilder().proxy(proxyOptions).build();


        TokenCredential certificateCredential = new ClientCertificateCredentialBuilder()
                .tenantId(tenantId)
                .clientId(clientId)
                .pemCertificate(TEMP_FILENAME)
                .httpClient(httpClient)
                .build();

        BlobServiceClient blobServiceClient = new BlobServiceClientBuilder()
                .endpoint(proxy_url)
                .credential(certificateCredential)
                .buildClient();

````
In the proxyOptions, we can only put the username/password, is there a possiblity to add the JWT in the headers as we used to do ?
We need help so that we know the right configuration to use the azure Java SDK
Thank you

**Setup (please complete the following information if applicable):**
 - OS: Linux
 - IDE:IntelliJ
 - Library/Libraries: [e.g. com.azure:azure-core:1.16.0 (groupId:artifactId:version)]
 azure-core--1.43.0
azure-identity--1.10.1
 **Information Checklist**
 Kindly make sure that you have added all the following information above and checkoff the required fields otherwise we will treat the issuer as an incomplete report
- [ ] Query Added
- [ ] Setup information Added

**Comment by joshfree** [comment]
@alzimmermsft this old issue seems to have fallen through the cracks; can you please double check / close as appropriate

**Comment by alzimmermsft** [comment]
Hi @zkhezane, thanks for filing this issue. At this time the only authentications available for proxying are `Basic` and `Digest`.

I'm going to change this over to a feature request and have a discussion with the team on allowing proxy authentication to be pluggable, ex. being able to be provided from an external source such as this case for JWT authentication or NTLM authentication.

**Comment by anuchandy** [comment]
Here is a [setup](https://github.com/anuchandy/az-java-proxy-sample) for Kerberos proxy server/client. Linking it here, so if needed this setup may be used to validate the adaptability of pluggable proxy contract just for testing purposes.

**Comment by erdeminanc** [comment]
> Here is a [setup](https://github.com/anuchandy/az-java-proxy-sample) for Kerberos proxy server/client. Linking it here, so if needed this setup may be used to validate the adaptability of pluggable proxy contract just for testing purposes.
> 
> For anyone referring to the setup: Please be aware that this is just for testing purposes, and we cannot guarantee that the settings we use in the example are sufficiently secure. I'm not a specialist in details of this security protocol. Consult expert for the correct and secure use in real world scenarios. The sample is just to show a basic concept of how challenges work in Kerberos, so we can play around.


Thanks for your prompt response.

The setup you have provided above shows creation of HttpClient, however EventProcessorClient is making Websocket connection. How should we configure Kerberos Authentication to proxy in this case?

**Comment by anuchandy** [comment]
@erdeminanc, right, the setup demonstrates Apache http client, I think that’s the only http client out there with built in Kerberos proxy auth. 

Here we are discussing about a Proxy interface/contract _feature_ to enable users to implement that interface for a specific auth of their interest (Kerberos, ntlm, jwt or any other) and plug it in. So sdk delegates proxy auth negotiation responsibility to the user provided implementation (for both http and websocket initiated via http).
